store/store.go

```diff
@@ -1124,7 +1147,7 @@ var download = func(name, sha3_384, downloadURL string, user *auth.UserState, s
 		// connections) that led us to an error (the default client is
 		// documented as not reusing the transport unless the body is
 		// read to EOF and closed, so this is a belt-and-braces thing).
-		r, err := s.doRequest(&http.Client{}, reqOptions, user)
+		r, err := s.doRequest(newHTTPClient(0), reqOptions, user)
```

Comment:
Actually after merging your branch I think its fine if we make it an explicit option (like you did there). Please let me know if you prefer the auto-detect approach, both is fine with me, I'm slightly on the fence but lean towards explicit.